CMakeLists.txt

```diff
@@ -249,30 +244,40 @@ else() # Standalone build
 	# We build SIMD filed separately, because they will be reached only if the
 	# machine running the code has SIMD support
 	set(SIMD_SOURCE_FILE "${PROJECT_SOURCE_DIR}/src/simulators/statevector/qv_avx2.cpp")
-	if(APPLE OR UNIX)
-		set_source_files_properties(${SIMD_SOURCE_FILE} PROPERTIES COMPILE_FLAGS "-mfma -mavx2")
-	elseif(MSVC)
-		set_source_files_properties(${SIMD_SOURCE_FILE} PROPERTIES COMPILE_FLAGS "/arch:AVX2")
-	endif()
-	set(AER_SIMULATOR_SOURCES "${PROJECT_SOURCE_DIR}/contrib/standalone/qasm_simulator.cpp"
-		"${SIMD_SOURCE_FILE}")
-	set_source_files_properties(${AER_SIMULATOR_SOURCES} PROPERTIES CUDA_SOURCE_PROPERTY_FORMAT OBJ)
 
+	set(AER_SIMULATOR_SOURCES "${PROJECT_SOURCE_DIR}/contrib/standalone/qasm_simulator.cpp")
 	set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 	if(CUDA_FOUND)
-		cuda_add_executable(qasm_simulator ${AER_SIMULATOR_SOURCES})
+		set_source_files_properties(${SIMD_SOURCE_FILE} PROPERTIES LANGUAGE CUDA)
```

Comment:
hmmm this is somhow a special file, not sure what would imply to set the `LANGUAGE` to `CUDA` here, but we must be extra careful with everything involving this file.

Reply:
This is the new general way to do the same as `set_source_files_properties(.. PROPERTIES CUDA_SOURCE_PROPERTY_FORMAT OBJ)` (how FindCUDA did it).